ocs/ocp.py

```diff
@@ -0,0 +1,106 @@
+"""
+General OCP object
+"""
+import os
+import logging
+import yaml
+
+from munch import munchify
+from utility.utils import TimeoutSampler
+from utility.utils import run_cmd
+
+log = logging.getLogger(__name__)
+
+
+class OCP(object):
+    """
+    TBD
+    """
+
+    def __init__(self, api_version='v1', kind='Service', namespace=None):
+        """
+        TBD
+
+        Args:
+            api_version (string): TBD
+            kind (string): TBD
+        """
+        self._api_version = api_version
+        self._kind = kind
+        self._namespace = namespace
+
+    @property
+    def api_version(self):
+        return self._api_version
+
+    @property
+    def kind(self):
+        return self._kind
+
+    @property
+    def namespace(self):
+        return self._namespace
+
+    def exec_oc_cmd(self, command, ):
+        """
+        TBD
+        """
+        oc_cmd = "oc "
+        kubeconfig = os.getenv('KUBECONFIG')
+        if self.namespace:
+            oc_cmd += f"-n {self.namespace} "
+
+        if kubeconfig:
+            oc_cmd += f"--kubeconfig {kubeconfig} "
+
+        oc_cmd += command
+        out = run_cmd(cmd=oc_cmd)
```

Comment:
in case of a failure, I believe that the output won't digestible munchify(yaml.safe_load(out))

Reply:
in a failure, an exception will raise from run_cmd so it won't reach munchify

Reply:
Right :)